syz-manager/hub.go

```diff
@@ -50,6 +53,21 @@ func (mgr *Manager) hubSyncLoop(keyGet keyGetter) {
 	if mgr.cfg.Reproduce && mgr.dash != nil {
 		hc.needMoreRepros = mgr.needMoreRepros
 	}
+
+	go func() {
+		time.Sleep(abandonSyzHub)
+		if hc.loadedOnce.Load() {
+			return
+		}
+		// After so many attempts syz-hub is still unreachable.
+		// We can continue operating as if it were never specified.
+		if mgr.dash != nil {
+			mgr.dash.LogError(mgr.cfg.Name, "did not manage to connect to syz-hub")
+		}
+		mgr.mu.Lock()
+		defer mgr.mu.Unlock()
+		mgr.phase = phaseTriagedHub
```

Comment:
This code is active even if we have not hub at all.
We also don't necessary triaged corpus within an hour.
We also need to check for hub sync rather than connect, because sync causes switch to phaseTriagedHub state.

I think this better belongs to HubConnector.loop. Say, it fails to sync 3 times, then it notifies manager to switch to phaseTriagedHub state.
Then it will be clear why we are promoting state and the state machine will be more deterministic.
We will also be able to log the hub error to the dashboard.
It may need some refactoring of Manager.hubSyncLoop and HubConnector.loop.

Reply:
> This code is active even if we have not hub at all.
> We also don't necessary triaged corpus within an hour.

https://github.com/google/syzkaller/blob/62df2017e3b1edd786a4c737bd4ccba2b4581d88/syz-manager/manager.go#L1381-L1383

We only invoke the code after corpus is triaged and only if the hub is present. Or am I missing something?

Reply:
You are right. For some reason I assumed the goroutine is started during manager startup.
Still looks more reasonable to do synchronously and deterministically rather than as an asynchronous patch.

Reply:
I've updated the code. Does it look better now?